### PR TITLE
source-incident-io contribution from aazam-gh

### LINKED ISSUE #45999: [connector-request] Incident.io
This is a **suggestion** to create a connector for **Incident.io** using the UI Connector Builder.
The website for the connector is https://incident.io.

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to 3 step)
3. Describe steps to get credentials.
4. Create the connector using the UI Builder

> [!IMPORTANT]
> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> [!CAUTION]
> This is a **suggestion** and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend to follow steps above.

### PR DESCRIPTION
**Pull request by aazam-gh**
## What

This PR adds a new connector for source Incident (source-incident).

Contributor-provided connector description:

Source connector for Incident which is an on all, incident response and status pages tool. 
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `api_key` | `string` | API Key. API key to use. Find it at https://app.incident.io/settings/api-keys |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| actions | id | No pagination | ✅ |  ❌  |
| catalog_types | id | No pagination | ✅ |  ❌  |
| custom_fields | id | No pagination | ✅ |  ❌  |
| follow-ups | id | No pagination | ✅ |  ❌  |
| incident_roles | id | No pagination | ✅ |  ❌  |
| incident_timestamps | id | No pagination | ✅ |  ❌  |
| incident_updates | id | DefaultPaginator | ✅ |  ❌  |
| incident_statuses | id | No pagination | ✅ |  ❌  |
| workflows | id | No pagination | ✅ |  ❌  |
| users | id | DefaultPaginator | ✅ |  ❌  |
| severities | id | No pagination | ✅ |  ❌  |
| schedules | id | DefaultPaginator | ✅ |  ❌  |
| incidents | id | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

**Comment by natikgadzhi** [comment]
I think we already have incident.io source, cc @marcosmarxm

**Comment by marcosmarxm** [comment]
> I think we already have incident.io source, cc @marcosmarxm

We don't have yet

File: docs/integrations/sources/incident.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Incident
+Source connector for Incident which is an on all, incident response and status pages tool.
```
Comment (marcosmarxm) [inline docs/integrations/sources/incident.md:2]: Please improve the description. Include url to api docs and service website.

File: docs/integrations/sources/incident.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+| follow-ups | id | No pagination | ✅ |  ❌  |
+| incident_roles | id | No pagination | ✅ |  ❌  |
+| incident_timestamps | id | No pagination | ✅ |  ❌  |
+| incident_updates | id | DefaultPaginator | ✅ |  ❌  |
+| incident_statuses | id | No pagination | ✅ |  ❌  |
+| workflows | id | No pagination | ✅ |  ❌  |
+| users | id | DefaultPaginator | ✅ |  ❌  |
+| severities | id | No pagination | ✅ |  ❌  |
+| schedules | id | DefaultPaginator | ✅ |  ❌  |
+| incidents | id | DefaultPaginator | ✅ |  ❌  |
```
Comment (marcosmarxm) [inline docs/integrations/sources/incident.md:25]: Some endpoints have support to use incremental, do you mind implement them?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments.

**Comment by natikgadzhi** [comment]
Huh! Alright, let's name it `source-incident-io` then, they're very recognizable with that name specifically. /cc @marcosmarxm

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Contribution looks great @aazam-gh can you just rename to `incident-io` to make clear what service the connector is about? Thanks

File: docs/integrations/sources/incident-io.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Incident.io
+
+This is the setup guide for the Incident.io source connector which ingests data from the incident API.
+
+## Prerequisites
+
+An API key is required for authentication and using this connector. In order to obtain an API key, you must first create an Incident.io account.
+You can create an account here https://incident.io/
```
Comment (natikgadzhi) [inline docs/integrations/sources/incident-io.md:8]: Linking to the vendor home page without clarifying how to navigate to the page where I can make a token is not good enough.

File: docs/integrations/sources/incident-io.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# Incident.io
+
+This is the setup guide for the Incident.io source connector which ingests data from the incident API.
```
Comment (natikgadzhi) [inline docs/integrations/sources/incident-io.md:3]: This sentence is not clear enough. What are the two most useful streams? 

Calling this a guide is not necessary. It's pretty obvious it's a guide from where we render this page.

**Review by marcosmarxm — APPROVED** [review]
Thanks @aazam-gh !

**Pull request merged**